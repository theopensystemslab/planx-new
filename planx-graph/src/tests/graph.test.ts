import Graph from "../graph";

const graph = new Graph();

test("add a node with children", () => {
  graph.add(
    { text: "question", type: 100 },
    { children: [{ text: "a1" }, { text: "a2" }] }
  );

  expect(graph).toMatchSnapshot();
});

describe("updating a node", () => {
  test("add a field to a without affecting existing data", () => {
    graph.update("b", { foo: "bar" });

    expect(graph).toMatchSnapshot();
  });

  test("replace existing data", () => {
    graph.update("b", { foo: "bar2" });

    expect(graph).toMatchSnapshot();
  });

  test("remove existing data", () => {
    graph.update("b", { foo: null });

    expect(graph).toMatchSnapshot();
  });
});

test("move a node", () => {
  graph.move("c", { fromParent: "a", toBefore: "b" });

  expect(graph).toMatchSnapshot();
});

test("remove a child node", () => {
  graph.remove("c");

  expect(graph).toMatchSnapshot();
});

test("remove a node with children", () => {
  graph.remove("a");

  expect(graph).toMatchSnapshot();
});
