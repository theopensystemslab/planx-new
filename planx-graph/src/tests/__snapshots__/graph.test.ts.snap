// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add a node with children 1`] = `
Graph {
  "counter": 3,
  "idFunction": [Function],
  "nodes": Map {
    "_root" => Object {
      "edges": Array [
        "a",
      ],
    },
    "a" => Object {
      "data": Object {
        "text": "question",
      },
      "edges": Array [
        "b",
        "c",
      ],
      "type": 100,
    },
    "b" => Object {
      "data": Object {
        "text": "a1",
      },
      "edges": Array [],
      "type": 200,
    },
    "c" => Object {
      "data": Object {
        "text": "a2",
      },
      "edges": Array [],
      "type": 200,
    },
  },
}
`;

exports[`move a node 1`] = `
Graph {
  "counter": 3,
  "idFunction": [Function],
  "nodes": Map {
    "_root" => Object {
      "edges": Array [
        "a",
      ],
    },
    "a" => Object {
      "data": Object {
        "text": "question",
      },
      "edges": Array [
        "c",
        "b",
      ],
      "type": 100,
    },
    "b" => Object {
      "data": Object {
        "text": "a1",
      },
      "edges": Array [],
      "type": 200,
    },
    "c" => Object {
      "data": Object {
        "text": "a2",
      },
      "edges": Array [],
      "type": 200,
    },
  },
}
`;

exports[`remove a child node 1`] = `
Graph {
  "counter": 3,
  "idFunction": [Function],
  "nodes": Map {
    "_root" => Object {
      "edges": Array [
        "a",
      ],
    },
    "a" => Object {
      "data": Object {
        "text": "question",
      },
      "edges": Array [
        "b",
      ],
      "type": 100,
    },
    "b" => Object {
      "data": Object {
        "text": "a1",
      },
      "edges": Array [],
      "type": 200,
    },
  },
}
`;

exports[`remove a node with children 1`] = `
Graph {
  "counter": 3,
  "idFunction": [Function],
  "nodes": Map {
    "_root" => Object {
      "edges": Array [],
    },
  },
}
`;

exports[`updating a node add a field to a without affecting existing data 1`] = `
Graph {
  "counter": 3,
  "idFunction": [Function],
  "nodes": Map {
    "_root" => Object {
      "edges": Array [
        "a",
      ],
    },
    "a" => Object {
      "data": Object {
        "text": "question",
      },
      "edges": Array [
        "b",
        "c",
      ],
      "type": 100,
    },
    "b" => Object {
      "data": Object {
        "foo": "bar",
        "text": "a1",
      },
      "edges": Array [],
      "type": 200,
    },
    "c" => Object {
      "data": Object {
        "text": "a2",
      },
      "edges": Array [],
      "type": 200,
    },
  },
}
`;

exports[`updating a node remove existing data 1`] = `
Graph {
  "counter": 3,
  "idFunction": [Function],
  "nodes": Map {
    "_root" => Object {
      "edges": Array [
        "a",
      ],
    },
    "a" => Object {
      "data": Object {
        "text": "question",
      },
      "edges": Array [
        "b",
        "c",
      ],
      "type": 100,
    },
    "b" => Object {
      "data": Object {
        "text": "a1",
      },
      "edges": Array [],
      "type": 200,
    },
    "c" => Object {
      "data": Object {
        "text": "a2",
      },
      "edges": Array [],
      "type": 200,
    },
  },
}
`;

exports[`updating a node replace existing data 1`] = `
Graph {
  "counter": 3,
  "idFunction": [Function],
  "nodes": Map {
    "_root" => Object {
      "edges": Array [
        "a",
      ],
    },
    "a" => Object {
      "data": Object {
        "text": "question",
      },
      "edges": Array [
        "b",
        "c",
      ],
      "type": 100,
    },
    "b" => Object {
      "data": Object {
        "foo": "bar2",
        "text": "a1",
      },
      "edges": Array [],
      "type": 200,
    },
    "c" => Object {
      "data": Object {
        "text": "a2",
      },
      "edges": Array [],
      "type": 200,
    },
  },
}
`;
